### ISSUE TITLE
unified-signatures incorrectly reports errors for defaulted paramters

### ISSUE THREAD

**Issue by Neos3452** [palantir/tslint#4059]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 2.9.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
function foo(num: number);
function foo(num: number, num1: number);
function foo(num: string);
function foo(num: number | string, num1: number = 0) {
    console.log(num);
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "unified-signatures": true
    }
}
```

#### Actual behavior
```
ERROR: index.ts[2, 27]: This overload and the one on line 1 can be combined into one signature with an optional parameter.
ERROR: index.ts[3, 14]: This overload and the one on line 1 can be combined into one signature taking `number | string`.
```
#### Expected behavior
No error.

**Comment by johnwiseheart** [comment]
The point of this rule is to create the minimal set of overloads. Because 
```typescript
function foo(num: number);
function foo(num: number, num1: number);
```

is equivalent to
```typescript
function foo(num: number, num1?: number);
```

you'll get an error here, which is expected.
